core/runtime/src/main/java/io/quarkus/runtime/graal/InetSubstitutions.java

```diff
@@ -0,0 +1,164 @@
+package io.quarkus.runtime.graal;
+
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+
+import org.wildfly.common.net.Inet;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.InjectAccessors;
+import com.oracle.svm.core.annotate.TargetClass;
+
+/*
+ * The following substitutions are required because of a new restriction in GraalVM 19.3.0 that prohibits the presence of
+ * java.net.Inet4Address and java.net.Inet6Address in the image heap. Each field annotated with @InjectAccessors is lazily
+ * recomputed at runtime on first access while Inet.class can still be initialized during the native image build.
+ */
+@TargetClass(Inet.class)
+final class Target_org_wildfly_common_net_Inet {
+
+    @Alias
+    @InjectAccessors(Inet4AnyAccessor.class)
+    public static Inet4Address INET4_ANY;
+
+    @Alias
+    @InjectAccessors(Inet4LoopbackAccessor.class)
+    public static Inet4Address INET4_LOOPBACK;
+
+    @Alias
+    @InjectAccessors(Inet4BroadcastAccessor.class)
+    public static Inet4Address INET4_BROADCAST;
+
+    @Alias
+    @InjectAccessors(Inet6AnyAccessor.class)
+    public static Inet6Address INET6_ANY;
+
+    @Alias
+    @InjectAccessors(Inet6LoopbackAccessor.class)
+    public static Inet6Address INET6_LOOPBACK;
+}
+
+class Inet4AnyAccessor {
```

Comment:
I think these accessors could be substantially simplified by introducing a new class, which is a run-time initialized class that includes the fields like this:
```java
    static final class Inet_RunTime {
        public static final Inet4Address INET4_ANY = Inet.getInet4Address(0, 0, 0, 0);
        public static final Inet4Address INET4_LOOPBACK = Inet.getInet4Address(127, 0, 0, 1);
        public static final Inet4Address INET4_BROADCAST = Inet.getInet4Address(255, 255, 255, 255);
        public static final Inet6Address INET6_ANY = Inet.getInet6Address(0, 0, 0, 0, 0, 0, 0, 0);
        public static final Inet6Address INET6_LOOPBACK = Inet.getInet6Address(0, 0, 0, 0, 0, 0, 0, 1);
    }
```

Then each accessor only has to look like this (for example):

```java
            static Inet4Address get() {
                return Inet_RunTime.INET4_LOOPBACK;
            }
```

and that would allow the accessor to be inlined, effectively performing a safe(r) partial class reinitialization at run time of that class.

Reply:
Simplifying the code was exactly what I was thinking about now that the reflection code is gone :) I'll try your suggestion right away, thanks!

Reply:
It seems GraalVM doesn't like this:
```
final class Inet_RunTime {
    public static final Inet4Address INET4_ANY = Inet.getInet4Address(0, 0, 0, 0);
    public static final Inet4Address INET4_LOOPBACK = Inet.getInet4Address(127, 0, 0, 1);
    public static final Inet4Address INET4_BROADCAST = Inet.getInet4Address(255, 255, 255, 255);
    public static final Inet6Address INET6_ANY = Inet.getInet6Address(0, 0, 0, 0, 0, 0, 0, 0);
    public static final Inet6Address INET6_LOOPBACK = Inet.getInet6Address(0, 0, 0, 0, 0, 0, 0, 1);
}
```
It is causing many errors locally including:
```
Error: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of java.net.Inet4Address are allowed in the image heap as this class should be initialized at image runtime. Object has been initialized without the native-image initialization instrumentation and the stack trace can't be tracked.
Trace: 
	at parsing io.quarkus.runtime.graal.Inet4AnyAccessor.get(InetSubstitutions.java:51)
```

Reply:
Oh wait, I didn't do anything about initializing the class at runtime. Trying again...

Reply:
Suggestion applied and working.